src/plugins/home/server/services/sample_data/data_sets/util.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SavedObject } from 'opensearch-dashboards/server';
+
+export const getSavedObjectsWithDataSource = (
+  saveObjectList: SavedObject[],
+  dataSourceId?: string,
+  dataSourceTitle?: string
+): SavedObject[] => {
+  if (dataSourceId) {
+    return saveObjectList.map((saveObject) => {
+      saveObject.id = `${dataSourceId}_` + saveObject.id;
```

Comment:
Is this `id` being saved in metadata? I am thinking should we just use a new uuid() here? So we are consistant in the format of id

Reply:
All the saved objects of sample data's ID, and all the reference are all hardcoded. This is the existing implementation.

Reply:
Also these sample data's id needs to have relationship with data sources id somehow. Thus I used the data source id to pre-append.